web/src/submessage.ts

```diff
@@ -204,11 +204,11 @@ export function make_server_callback(
     message_id: number,
 ): (opts: {
     msg_type: string;
-    data: string | PollWidgetOutboundData | TodoWidgetOutboundData;
+    data: string | PollWidgetOutboundData | TodoWidgetOutboundData | undefined;
```

Comment:
I feel think `undefined` is not valid here. If `zform` doesn't support outbound data, so I think we should instead be avoiding this function being called from that code path.

Reply:
Great point, fixed by not calling `activate` when it's falsey.

Reply:
I'm a little wary of doing that because I don't feel confident that we don't want to `make_server_callback` with `undefined` data. If you feel confident enough in making that change, I can add a commit for it to this PR and we can merge it.

Reply:
I am not confident of it at all. But I think you usually would want to not pass data at all over passing `undefined`? I am skeptical that this works properly.

Reply:
Can you give a specific situation where you can imagine passing `undefined` in one of these callbacks makes sense? It would seem that it'd make a callback with no data in the request.